crypto/dsa/dsa_ossl.c

```diff
@@ -126,6 +126,10 @@ static DSA_SIG *dsa_do_sign(const unsigned char *dgst, int dlen, DSA *dsa)
     BN_set_flags(tmp, BN_FLG_CONSTTIME);
 
     /* tmp := blind * priv_key * r mod q */
+    if (dsa->priv_key == NULL) {
+        reason = DSA_R_MISSING_PRIVATE_KEY;
+        goto err;
+    }
```

Comment:
I'd suggest moving this right up to near the check for other missing parameters: p, q and g

Reply:
Done